devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java

```diff
@@ -95,32 +69,17 @@ public void relink(AppArtifact appArtifact, Path localPath) throws AppModelResol
     @Override
     public Path resolve(AppArtifact appArtifact) throws AppModelResolverException {
         if (!appArtifact.isResolved()) {
-
-            final DefaultDependencyArtifact dep = new DefaultDependencyArtifact();
-            dep.setExtension(appArtifact.getType());
-            dep.setType(appArtifact.getType());
-            dep.setName(appArtifact.getArtifactId());
-
-            final DefaultExternalModuleDependency gradleDep = new DefaultExternalModuleDependency(appArtifact.getGroupId(),
-                    appArtifact.getArtifactId(), appArtifact.getVersion(), null);
-            gradleDep.addArtifact(dep);
-
-            final Configuration detachedConfig = project.getConfigurations().detachedConfiguration(gradleDep);
-
-            final ResolvedConfiguration rc = detachedConfig.getResolvedConfiguration();
-            Set<ResolvedArtifact> resolvedArtifacts = rc.getResolvedArtifacts();
-            for (ResolvedArtifact a : resolvedArtifacts) {
-                if (appArtifact.getArtifactId().equals(a.getName())
-                        && appArtifact.getType().equals(a.getType())
-                        && appArtifact.getGroupId().equals(a.getModuleVersion().getId().getGroup())) {
-                    appArtifact.setPath(a.getFile().toPath());
+            for (io.quarkus.gradle.workspace.model.Dependency dependency : model.getAppDependencies()) {
```

Comment:
This is actually very limiting, since the scope is limited to the app model only. I'll need to check whether the devtools could be affected.

Reply:
Have you tried running the full testsuite with these changes? Or more specifically all the gradle tests including the tooling ones, such add/list extensions, create project? I'm wondering whether we should be changing this method at all. What we need to providing at the end is the AppModel and the workspace structure.

Reply:
I run some tests on my machine on a sample project and everything was ok. Regarding the testsuite, I still have some failures. I will revert that change as you suggest.